packages/botonic-cli/src/botonic-api-service.ts

```diff
@@ -129,39 +129,56 @@ export class BotonicAPIService {
     if (pathExists(pathToCredentials)) unlinkSync(pathToCredentials)
   }
 
-  async api(
-    path: string,
-    body: any = null,
-    method: Method = 'get',
-    headers: any = null,
-    params: any = null
-  ): Promise<any> {
-    let b = 0
-    try {
-      return await axios({
-        method: method,
-        url: this.baseApiUrl + path,
+  async apiPost({
+    path,
+    body,
+    headers,
+    params = null,
+  }: {
+    path: string
+    body?: any
+    headers?: any
+    params?: any
+  }): Promise<any> {
+    const requestFn = () =>
+      axios.post(`${this.baseUrl}${path}`, body, {
+        headers: headers || this.headers,
+        params,
+      })
+
+    return this.retryOnUnauthorized(requestFn)
+  }
+
+  async apiGet({
+    path,
+    headers,
+    params = null,
+  }: {
+    path: string
+    headers?: any
+    params?: any
+  }): Promise<any> {
+    const requestFn = () =>
+      axios.get(`${this.baseUrl}${path}`, {
         headers: headers || this.headers,
-        data: body,
-        params: params,
+        params,
       })
-    } catch (e: any) {
-      if (e.response.status == 401) {
-        b = 1
+    return this.retryOnUnauthorized(requestFn)
```

Comment:
As we talked in the meeting, take a look at the interceptors.

Reply:
Changed in the next [PR#2894](https://github.com/hubtype/botonic/pull/2894)